packages/eas-cli/src/build/utils/appJson.ts

```diff
@@ -0,0 +1,25 @@
+// import { ExpoConfig, getConfigFilePaths } from '@expo/config';
+import * as ExpoConfig from '@expo/config';
+import assert from 'assert';
+import fs from 'fs-extra';
+
+export async function updateAppJsonConfigAsync(
+  {
+    projectDir,
+    exp,
+  }: {
+    projectDir: string;
+    exp: ExpoConfig.ExpoConfig;
+  },
+  modifyConfig: (config: any) => void
+): Promise<void> {
+  const paths = ExpoConfig.getConfigFilePaths(projectDir);
+  assert(paths.staticConfigPath, "can't update dynamic configs");
```

Comment:
Make sure that this condition is handled earlier with proper error message sth along the lines `autoIncrement` options is not supported with dynamic config

Also(optional) when you check that you could also verify if the  version in config is the same as in static config

Reply:
Done